src/components/CartProduct/index.tsx

```diff
@@ -0,0 +1,63 @@
+import { useDispatch } from "react-redux";
+import { CartItem, decrement, deleteItem, increment, selectItem } from "../../redux/modules/cart";
+import CheckBox from "../@shared/CheckBox/styles";
+import { INFO_MESSAGES, PRODUCT } from "../../constants";
+import Delete from "../../assets/Delete.png";
+import {
+  CartProductContainer,
+  ProductAmountContainer,
+  ProductAmountWrapper,
+  ProductCounterContainer,
+  ProductImageContainer,
+  ProductOptionContainer,
+} from "./styles";
+
+interface CartProductProps {
+  item: CartItem;
+}
+
+function CartProduct({ item: { id, img, name, price, amount, isSelected } }: CartProductProps) {
+  const dispatch = useDispatch();
+
+  const onToggleSelect = () => {
+    dispatch(selectItem(id));
+  };
+
+  const onClickDeleteItem = () => {
+    confirm(INFO_MESSAGES.ASK_DELETE_PRODUCT) && dispatch(deleteItem(id));
+  };
+
+  const onClickIncreaseCounter = () => {
+    dispatch(increment(id));
+  };
+
+  const onClickDecreaseCounter = () => {
+    if (amount === PRODUCT.MIN_COUNT) {
+      return;
+    }
+    dispatch(decrement(id));
+  };
```

Comment:
```suggestion
   const { increaseItem, descreaseItem, toggleItem  } = useCartItem({ id, quantity: amount });
```
요거 hook으로 분리해보면 어떨까요?
어떻게 하는지만 명시해봐도 좋을 것 같다고 생각이 들어서요~

Reply:
구체적인 구현을 숨길수있어서 좀더 깔끔한 방법인것같네요! 감사합니다!!